src/io/backend_cubeb.rs

```diff
@@ -195,48 +201,55 @@ impl AudioBackend for CubebBackend {
             .ok()
             .unwrap_or(RENDER_QUANTUM_SIZE as u32);
         let buffer_size = buffer_size_req.max(min_latency);
+        let device_id = options.sink_id.as_ref().and_then(|d| {
+            Self::enumerate_devices()
+                .into_iter()
+                .find(|e| e.device_id() == d)
+                .map(|e| *e.device().downcast::<DeviceId>().unwrap())
+        });
 
         let stream = match number_of_channels {
             // so sorry, but I need to constify the non-const `number_of_channels`
```

Comment:
I know you don't like these, but a macro would be appropriate here I think :), as it could generate the code according to MAX_CHANNELS

Reply:
You are probably right. But I don't have the expertise to come up with the macro declaration.
Reading https://stackoverflow.com/questions/33751796/is-there-a-way-to-count-with-macros this cannot be done easily with `macro_rules!`, and we would have to resort to procedural macros which would be really overkill.
I would love to be proven wrong though! Any ideas?